### DISCUSSION TITLE
Best way to pack program into an executable?

### DISCUSSION THREAD

**Discussion by Frityet** [keplerproject/luarocks#1455] [Q&A]
What are the best/most popular ways to pack a regular luarocks module into an executable that can be easily installed (like, a `.app` on MacOS, a `.exe` on windows)?

**Comment by alerque** [comment]
First of all, this really isn't a LuaRocks issue. This discussion should probably be taken elsewhere, StackOverflow for example.

As a pointer in the general direction, what you are looking for is not LuaRocks (which is specifically for packaging and distributing inside the Lua ecosystem) but something to work outside of the ecosystem. That probably means first bundling all your Lua stuff together in an amalgam, then using other tooling to combine a Lua interpreter to ship with your code. I know `luastatic` is one such tool for the last bit, and #1452 (also off topic here) touched on the first bit.

**Reply by Frityet** [reply]
I thought the discussions section could be used for non-isses. Thank you though!

**Reply by alerque** [reply]
Maybe, I'm not in charge of this project ;-) Surely the rules are looser for discussions than issues. But topically this isn't really about `luarocks` which is a package manager, it is conceptually about static compiling/bundling which is a different ball of wax.

**Reply by Frityet** [reply]
> Maybe, I'm not in charge of this project ;-) Surely the rules are looser for discussions than issues. But topically this isn't really about `luarocks` which is a package manager, it is conceptually about static compiling/bundling which is a different ball of wax.

Oh no I was asking here about how to also pack my luarocks modules into the binary aswell, but I understand.
